myhoard/backup_stream.py

```diff
@@ -75,6 +75,26 @@ class BaseBackupFailureReason(str, enum.Enum):
     xtrabackup_error = "xtrabackup_error"
 
 
+class BaseBackup(TypedDict, total=False):
```

Comment:
Just moved this from myhoard.Controller and added rest of expected fields.  The `total=False` is mainly in case a `basebackup.json` file is not containing all keys (ideally shouldn't happen but just in case)